internal/provider/resource_gmail_send_as_alias.go

```diff
@@ -0,0 +1,297 @@
+	}
+}
+
+func resourceGmailSendAsAliasCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*apiClient)
+
+	primaryEmail := d.Get("primary_email").(string)
+	gmailService, diags := client.NewGmailService(ctx, primaryEmail)
+	if diags.HasError() {
+		return diags
+	}
+
+	sendAsAliasService, diags := GetGmailSendAsAliasService(gmailService)
+	if diags.HasError() {
+		return diags
+	}
+
+	sendAsEmail := d.Get("send_as_email").(string)
+	log.Printf("[DEBUG] Creating Gmail Send As Alias %q", primaryEmail+sendAsIdSeparator+sendAsEmail)
+
+	sendAs, err := sendAsAliasService.Create("me", &gmail.SendAs{
+		SendAsEmail:    sendAsEmail,
+		DisplayName:    d.Get("display_name").(string),
+		ReplyToAddress: d.Get("reply_to_address").(string),
+		Signature:      d.Get("signature").(string),
+		IsDefault:      d.Get("is_default").(bool),
+		TreatAsAlias:   d.Get("treat_as_alias").(bool),
+		SmtpMsa:        expandSmtpMsa(d.Get("smtp_msa").([]interface{})),
+	}).Do()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.Set("send_as_email", sendAs.SendAsEmail)
+	d.SetId(primaryEmail + sendAsIdSeparator + sendAs.SendAsEmail)
+
+	log.Printf("[DEBUG] Finished creating Gmail Send As Alias %q", d.Id())
+
+	return resourceGmailSendAsAliasRead(ctx, d, meta)
+}
+
+func resourceGmailSendAsAliasUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*apiClient)
+
+	primaryEmail := d.Get("primary_email").(string)
+	gmailService, diags := client.NewGmailService(ctx, primaryEmail)
+	if diags.HasError() {
+		return diags
+	}
+
+	sendAsAliasService, diags := GetGmailSendAsAliasService(gmailService)
+	if diags.HasError() {
+		return diags
+	}
+
+	log.Printf("[DEBUG] Updating Gmail Send As Alias %q", d.Id())
+
+	_, err := sendAsAliasService.Update("me", d.Get("send_as_email").(string), &gmail.SendAs{
+		DisplayName:    d.Get("display_name").(string),
+		ReplyToAddress: d.Get("reply_to_address").(string),
+		Signature:      d.Get("signature").(string),
+		IsDefault:      d.Get("is_default").(bool),
+		TreatAsAlias:   d.Get("treat_as_alias").(bool),
+		SmtpMsa:        expandSmtpMsa(d.Get("smtp_msa").([]interface{})),
+	}).Do()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	log.Printf("[DEBUG] Finished updating Gmail Send As Alias %q", d.Id())
+
+	return resourceGmailSendAsAliasRead(ctx, d, meta)
+}
+
+func resourceGmailSendAsAliasRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*apiClient)
+
+	primaryEmail := d.Get("primary_email").(string)
+	gmailService, diags := client.NewGmailService(ctx, primaryEmail)
+	if diags.HasError() {
+		return diags
+	}
+
+	sendAsAliasService, diags := GetGmailSendAsAliasService(gmailService)
+	if diags.HasError() {
+		return diags
+	}
+
+	log.Printf("[DEBUG] Getting Gmail Send As Alias %q", d.Id())
+
+	sendAs, err := sendAsAliasService.Get("me", d.Get("send_as_email").(string)).Do()
```

Comment:
Is "me" just a placeholder here?

Reply:
No, since we impersonated the user getting the changes, you can put "me" as a special "userId".. Ideally this would just work without the impersonation workaround